src/core/include/iDynTree/Core/VectorFixSize.h

```diff
@@ -176,6 +185,13 @@ namespace iDynTree
         return VecSize;
     }
 
+    template<unsigned int VecSize>
+    VectorFixSize<VecSize> & VectorFixSize<VecSize>::operator=(const Span<const double>& vec) {
+        assert(VecSize == vec.size());
```

Comment:
Not sure is this is safe, but is probably the best option without resorting to exceptions.

Reply:
I was thinking to add an error message, but I would have preferred to avoid using an `if` there.

Reply:
The nice thing is that being in the header, if the user code is compiled in debug, the assert is used, even if iDynTree is compiled in Release.